store/localcachelayer/user_layer.go

```diff
@@ -39,6 +49,48 @@ func (s LocalCacheUserStore) InvalidatProfileCacheForUser(userId string) {
 	}
 }
 
+func (s LocalCacheUserStore) InvalidateProfilesInChannelCacheByUser(userId string) {
+	keys := s.rootStore.profilesInChannelCache.Keys()
+
+	for _, key := range keys {
+		if cacheItem, ok := s.rootStore.profilesInChannelCache.Get(key); ok {
+			userMap := cacheItem.(map[string]*model.User)
+			if _, userInCache := userMap[userId]; userInCache {
+				s.rootStore.doInvalidateCacheCluster(s.rootStore.profilesInChannelCache, key.(string))
+				if s.rootStore.metrics != nil {
+					s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Profiles in Channel - Remove by User")
+				}
+			}
+		}
+	}
+}
+
+func (s LocalCacheUserStore) InvalidateProfilesInChannelCache(channelId string) {
+	s.rootStore.doInvalidateCacheCluster(s.rootStore.profilesInChannelCache, channelId)
+	if s.rootStore.metrics != nil {
```

Comment:
Same here about the metrics, is handled by the doInvalidateCacheCluster (I think).